src/asl.ts

```diff
@@ -5261,36 +5270,41 @@ export namespace ASLGraph {
       operator === ">=" ||
       operator === "<="
     ) {
+      // left is a condition and right is any value
       if (ASLGraph.isConditionOutput(left)) {
         return (
           ASLGraph.booleanCompare(left, right, operator) ?? ASL.falseCondition()
         );
       }
+
+      // returns comparisons if there are valid comparisons to return.
+      // comparisons return undefined when the right is a literal and not the right
+      // type to compare (aString === aNumber) or when the operator doesn't make sense for the type (ex: less than boolean)
+      // when L and R are paths: typeof L === "string" && stringCompare(L, R)
       const condition = ASL.or(
         ASLGraph.nullCompare(left, right, operator),
         ASLGraph.stringCompare(left, right, operator),
         ASLGraph.booleanCompare(left, right, operator),
         ASLGraph.numberCompare(left, right, operator)
       );
 
-      if (ASLGraph.isJsonPath(right)) {
-        ASL.or(
-          // a === b while a and b are both not defined
-          ASL.not(
-            ASL.and(ASL.isPresent(left.jsonPath), ASL.isPresent(right.jsonPath))
-          ),
-          // a !== undefined && b !== undefined
-          ASL.and(
-            ASL.isPresent(left.jsonPath),
-            ASL.isPresent(right.jsonPath),
-            // && a [op] b
-            condition
-          )
-        );
-      }
-      return ASL.and(ASL.isPresent(left.jsonPath), condition);
+      return ASL.or(
+        // a === b while a and b are both not defined
+        ASLGraph.undefinedCompare(left, right, operator),
+        // a !== undefined && b !== undefined
+        ASL.and(
```

Comment:
I don't understand what the comments mean

Reply:
Woops, not relevant anymore.